pkg/epp/scheduling/types/types.go

```diff
@@ -71,16 +69,7 @@ func (pm *PodMetrics) GetMetrics() *backendmetrics.Metrics {
 	return pm.Metrics
 }
 
-func (pm *PodMetrics) SetScore(score float64) {
-	pm.score = score
-}
-
-func (pm *PodMetrics) Score() float64 {
-	return pm.score
-}
-
 type PodMetrics struct {
-	score float64
```

Comment:
I will keep the score here. It's clean, extensible,  and also allows sorting pods based on score easier, than a map from pods to scores.

Reply:
score is not a property of PodMetrics snapshot, but a result of calling the Score function.
PodMetrics is a snapshot of metrics at a given point in time and should be a read only interface that helps in plugins calculations during the scheduling cycle (e.g., filter/scorer may used metrics).

we may add a new, different struct, `ScoredPod` to have the fields `Pod` and `Score` and use that instead of the map, which will allow the easy sorting. 
```go
type ScoredPod struct {
    Pod types.Pod
    Score float64
}
```

LMKWYT

Reply:
Sg, I prefer []ScoredPod than the map, since it makes either for random picking or sorting based on score easier. Another option is to have `ScoredPods` which can contain both a list and a map, if the map is needed for any reason in the future.

Reply:
I added the `ScoredPod` struct  and updated the code to use it.
please pay attention that `Score` function still defines the return value as map[pod]float which is exactly the definition of a scorer, giving a score to each of the candidate pods.
this map is also very useful for accumulating the weighted score per pod in a fast way (hash). 

before passing the pods with the weighted scores to the picker, this is converted into []ScoredPod.

the way I see it, picker needs to be able to iterate over the pods AND to be able to sort pods based on score.
no use of both slice and map at this point. but if it becomes relevant at some point we can add.